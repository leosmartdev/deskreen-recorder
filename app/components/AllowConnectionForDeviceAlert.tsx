import React from 'react';
import { Intent, Alert, H4 } from '@blueprintjs/core';
import { useTranslation } from 'react-i18next';
import DeviceInfoCallout from './DeviceInfoCallout';
import isWithReactRevealAnimations from '../utils/isWithReactRevealAnimations';

interface AllowConnectionForDeviceAlertProps {
  device: Device | null;
  isOpen: boolean;
  onCancel: () => void;
  onConfirm: () => void;
}

export default function AllowConnectionForDeviceAlert(
  props: AllowConnectionForDeviceAlertProps
) {
  const { t } = useTranslation();
  const { device, isOpen, onCancel, onConfirm } = props;
  const denyText = t('Deny');
  const allowText = t('Allow');

  return (
    <Alert
      className="class-allow-device-to-connect-alert"
      cancelButtonText={denyText}
      confirmButtonText={allowText}
      icon="feed"
      intent={Intent.DANGER}
      isOpen={isOpen}
      onCancel={onCancel}
      onConfirm={onConfirm}
      transitionDuration={isWithReactRevealAnimations() ? 700 : 0}
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-ignore
      usePortal={false}
    >
      <H4>{t('Someone is trying to connect, do you allow?')}</H4>
      <DeviceInfoCallout
        deviceType={device?.deviceType}
        deviceIP={device?.deviceIP}
        deviceOS={device?.deviceOS}
        sharingSessionID={device?.sharingSessionID}
        deviceBrowser={device?.deviceBrowser}
      />
    </Alert>
  );
}
