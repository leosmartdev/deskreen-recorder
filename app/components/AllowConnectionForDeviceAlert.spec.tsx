import React from 'react';
import Enzyme, { mount } from 'enzyme';
import EnzymeToJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter as Router } from 'react-router-dom';
import AllowConnectionForDeviceAlert from './AllowConnectionForDeviceAlert';

Enzyme.configure({ adapter: new Adapter() });
jest.useFakeTimers();

it('should match exact snapshot', () => {
  const subject = mount(
    <>
      <Router>
        <AllowConnectionForDeviceAlert
          device={{} as Device}
          isOpen
          onCancel={() => {}}
          onConfirm={() => {}}
        />
      </Router>
    </>
  );
  expect(EnzymeToJson(subject)).toMatchSnapshot();
});
