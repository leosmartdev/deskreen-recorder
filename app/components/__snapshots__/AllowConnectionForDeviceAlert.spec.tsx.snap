// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match exact snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <AllowConnectionForDeviceAlert
      device={Object {}}
      isOpen={true}
      onCancel={[Function]}
      onConfirm={[Function]}
    >
      <Blueprint3.Alert
        canEscapeKeyCancel={false}
        canOutsideClickCancel={false}
        cancelButtonText=""
        className="class-allow-device-to-connect-alert"
        confirmButtonText=""
        icon="feed"
        intent="danger"
        isOpen={true}
        onCancel={[Function]}
        onConfirm={[Function]}
        transitionDuration={0}
        usePortal={false}
      >
        <Blueprint3.Dialog
          canEscapeKeyClose={false}
          canOutsideClickClose={false}
          className="bp3-alert class-allow-device-to-connect-alert"
          isOpen={true}
          onCancel={[Function]}
          onClose={[Function]}
          onConfirm={[Function]}
          transitionDuration={0}
          usePortal={false}
        >
          <Blueprint3.Overlay
            autoFocus={true}
            backdropProps={Object {}}
            canEscapeKeyClose={false}
            canOutsideClickClose={false}
            className="bp3-overlay-scroll-container"
            enforceFocus={true}
            hasBackdrop={true}
            isOpen={true}
            lazy={true}
            onCancel={[Function]}
            onClose={[Function]}
            onConfirm={[Function]}
            transitionDuration={0}
            transitionName="bp3-overlay"
            usePortal={false}
          >
            <TransitionGroup
              appear={true}
              childFactory={[Function]}
              className="bp3-overlay bp3-overlay-open bp3-overlay-inline bp3-overlay-scroll-container"
              component="div"
              onKeyDown={[Function]}
            >
              <div
                className="bp3-overlay bp3-overlay-open bp3-overlay-inline bp3-overlay-scroll-container"
                onKeyDown={[Function]}
              >
                <CSSTransition
                  addEndListener={[Function]}
                  appear={true}
                  classNames="bp3-overlay"
                  in={true}
                  key=".$__backdrop"
                  onExited={[Function]}
                  timeout={0}
                >
                  <Transition
                    addEndListener={[Function]}
                    appear={true}
                    enter={true}
                    exit={true}
                    in={true}
                    mountOnEnter={false}
                    onEnter={[Function]}
                    onEntered={[Function]}
                    onEntering={[Function]}
                    onExit={[Function]}
                    onExited={[Function]}
                    onExiting={[Function]}
                    timeout={0}
                    unmountOnExit={false}
                  >
                    <div
                      className="bp3-overlay-backdrop"
                      onMouseDown={[Function]}
                    />
                  </Transition>
                </CSSTransition>
                <CSSTransition
                  addEndListener={[Function]}
                  appear={true}
                  classNames="bp3-overlay"
                  in={true}
                  key=".$.0"
                  onExited={[Function]}
                  timeout={0}
                >
                  <Transition
                    addEndListener={[Function]}
                    appear={true}
                    enter={true}
                    exit={true}
                    in={true}
                    mountOnEnter={false}
                    onEnter={[Function]}
                    onEntered={[Function]}
                    onEntering={[Function]}
                    onExit={[Function]}
                    onExited={[Function]}
                    onExiting={[Function]}
                    timeout={0}
                    unmountOnExit={false}
                  >
                    <div
                      className="bp3-dialog-container bp3-overlay-content"
                      tabIndex={0}
                    >
                      <div
                        className="bp3-dialog bp3-alert class-allow-device-to-connect-alert"
                      >
                        <div
                          className="bp3-alert-body"
                        >
                          <Blueprint3.Icon
                            icon="feed"
                            iconSize={40}
                            intent="danger"
                          >
                            <span
                              className="bp3-icon bp3-icon-feed bp3-intent-danger"
                              icon="feed"
                            >
                              <svg
                                data-icon="feed"
                                height={40}
                                viewBox="0 0 20 20"
                                width={40}
                              >
                                <desc>
                                  feed
                                </desc>
                                <path
                                  d="M2.5 15a2.5 2.5 0 000 5 2.5 2.5 0 000-5zm.5-5c-.55 0-1 .45-1 1s.45 1 1 1c2.76 0 5 2.24 5 5 0 .55.45 1 1 1s1-.45 1-1c0-3.87-3.13-7-7-7zM3 0c-.55 0-1 .45-1 1s.45 1 1 1c8.28 0 15 6.72 15 15 0 .55.45 1 1 1s1-.45 1-1C20 7.61 12.39 0 3 0zm0 5c-.55 0-1 .45-1 1s.45 1 1 1c5.52 0 10 4.48 10 10 0 .55.45 1 1 1s1-.45 1-1C15 10.37 9.63 5 3 5z"
                                  fillRule="evenodd"
                                  key="0"
                                />
                              </svg>
                            </span>
                          </Blueprint3.Icon>
                          <div
                            className="bp3-alert-contents"
                          >
                            <Component>
                              <h4
                                className="bp3-heading"
                              />
                            </Component>
                            <DeviceInfoCallout>
                              <Component
                                style={
                                  Object {
                                    "margin": "0 auto",
                                    "textAlign": "center",
                                  }
                                }
                              >
                                <h4
                                  className="bp3-heading"
                                  style={
                                    Object {
                                      "margin": "0 auto",
                                      "textAlign": "center",
                                    }
                                  }
                                >
                                  :
                                </h4>
                              </Component>
                              <Blueprint3.Callout
                                id="device-info-callout"
                                style={
                                  Object {
                                    "borderRadius": "8px",
                                  }
                                }
                              >
                                <div
                                  className="bp3-callout"
                                  id="device-info-callout"
                                  style={
                                    Object {
                                      "borderRadius": "8px",
                                    }
                                  }
                                >
                                  <Row
                                    center="xs"
                                  >
                                    <div
                                      className="row center-xs"
                                    >
                                      <Col
                                        xs={12}
                                      >
                                        <div
                                          className="col-xs-12"
                                        >
                                          <Blueprint3.Text
                                            ellipsize={false}
                                            tagName="div"
                                          >
                                            <div
                                              className=""
                                            >
                                              :
                                               
                                              <span />
                                            </div>
                                          </Blueprint3.Text>
                                          <Blueprint3.Tooltip
                                            content={
                                              <React.Fragment>
                                                <Blueprint3.Text
                                                  ellipsize={false}
                                                  tagName="div"
                                                >
                                                  
                                                </Blueprint3.Text>
                                                <span
                                                  style={
                                                    Object {
                                                      "fontWeight": 900,
                                                    }
                                                  }
                                                >
                                                  <Blueprint3.Text
                                                    ellipsize={false}
                                                    tagName="div"
                                                  >
                                                    
                                                  </Blueprint3.Text>
                                                </span>
                                              </React.Fragment>
                                            }
                                            hoverCloseDelay={0}
                                            hoverOpenDelay={100}
                                            minimal={false}
                                            position="top"
                                            transitionDuration={100}
                                          >
                                            <Blueprint3.Popover
                                              autoFocus={false}
                                              boundary="scrollParent"
                                              canEscapeKeyClose={false}
                                              captureDismiss={false}
                                              content={
                                                <React.Fragment>
                                                  <Blueprint3.Text
                                                    ellipsize={false}
                                                    tagName="div"
                                                  >
                                                    
                                                  </Blueprint3.Text>
                                                  <span
                                                    style={
                                                      Object {
                                                        "fontWeight": 900,
                                                      }
                                                    }
                                                  >
                                                    <Blueprint3.Text
                                                      ellipsize={false}
                                                      tagName="div"
                                                    >
                                                      
                                                    </Blueprint3.Text>
                                                  </span>
                                                </React.Fragment>
                                              }
                                              defaultIsOpen={false}
                                              disabled={false}
                                              enforceFocus={false}
                                              fill={false}
                                              hasBackdrop={false}
                                              hoverCloseDelay={0}
                                              hoverOpenDelay={100}
                                              inheritDarkTheme={true}
                                              interactionKind="hover-target"
                                              lazy={true}
                                              minimal={false}
                                              modifiers={
                                                Object {
                                                  "arrow": Object {
                                                    "enabled": true,
                                                  },
                                                }
                                              }
                                              openOnTargetFocus={true}
                                              popoverClassName="bp3-tooltip"
                                              position="top"
                                              targetTagName="span"
                                              transitionDuration={100}
                                              usePortal={true}
                                              wrapperTagName="span"
                                            >
                                              <Manager>
                                                <span
                                                  className="bp3-popover-wrapper"
                                                >
                                                  <Reference
                                                    innerRef={[Function]}
                                                  >
                                                    <InnerReference
                                                      innerRef={[Function]}
                                                      setReferenceNode={[Function]}
                                                    >
                                                      <Blueprint3.ResizeSensor
                                                        onResize={[Function]}
                                                      >
                                                        <span
                                                          className="bp3-popover-target"
                                                          onBlur={[Function]}
                                                          onFocus={[Function]}
                                                          onMouseEnter={[Function]}
                                                          onMouseLeave={[Function]}
                                                        >
                                                          <div
                                                            className=""
                                                            key=".0"
                                                            style={
                                                              Object {
                                                                "backgroundColor": "#00f99273",
                                                                "borderRadius": "20px",
                                                                "fontWeight": 900,
                                                                "paddingLeft": "10px",
                                                                "paddingRight": "10px",
                                                              }
                                                            }
                                                            tabIndex={0}
                                                          >
                                                            <Blueprint3.Text
                                                              className="bp3-text-large"
                                                              ellipsize={false}
                                                              tagName="div"
                                                            >
                                                              <div
                                                                className="bp3-text-large"
                                                              >
                                                                :
                                                                 
                                                                <span
                                                                  className="device-ip-span"
                                                                />
                                                              </div>
                                                            </Blueprint3.Text>
                                                          </div>
                                                        </span>
                                                      </Blueprint3.ResizeSensor>
                                                    </InnerReference>
                                                  </Reference>
                                                  <Blueprint3.Overlay
                                                    autoFocus={false}
                                                    backdropClassName="bp3-popover-backdrop"
                                                    backdropProps={Object {}}
                                                    canEscapeKeyClose={false}
                                                    canOutsideClickClose={false}
                                                    enforceFocus={false}
                                                    hasBackdrop={false}
                                                    isOpen={false}
                                                    lazy={true}
                                                    onClose={[Function]}
                                                    transitionDuration={100}
                                                    transitionName="bp3-popover"
                                                    usePortal={true}
                                                  />
                                                </span>
                                              </Manager>
                                            </Blueprint3.Popover>
                                          </Blueprint3.Tooltip>
                                          <Blueprint3.Text
                                            ellipsize={false}
                                            tagName="div"
                                          >
                                            <div
                                              className=""
                                            >
                                              :
                                               
                                              <span />
                                            </div>
                                          </Blueprint3.Text>
                                          <Blueprint3.Text
                                            ellipsize={false}
                                            tagName="div"
                                          >
                                            <div
                                              className=""
                                            >
                                              :
                                               
                                              <span />
                                            </div>
                                          </Blueprint3.Text>
                                          <div
                                            style={
                                              Object {
                                                "margin": "0 auto",
                                                "width": "200px",
                                              }
                                            }
                                          >
                                            <Blueprint3.Text
                                              className="bp3-text-muted"
                                              ellipsize={true}
                                              tagName="div"
                                            >
                                              <div
                                                className="bp3-text-overflow-ellipsis bp3-text-muted"
                                              >
                                                :
                                                 
                                                <span />
                                              </div>
                                            </Blueprint3.Text>
                                          </div>
                                        </div>
                                      </Col>
                                    </div>
                                  </Row>
                                </div>
                              </Blueprint3.Callout>
                            </DeviceInfoCallout>
                          </div>
                        </div>
                        <div
                          className="bp3-alert-footer"
                        >
                          <Blueprint3.Button
                            intent="danger"
                            onClick={[Function]}
                            text=""
                          >
                            <button
                              className="bp3-button bp3-intent-danger"
                              onBlur={[Function]}
                              onClick={[Function]}
                              onKeyDown={[Function]}
                              onKeyUp={[Function]}
                              type="button"
                            >
                              <Blueprint3.Icon
                                key="leftIcon"
                              />
                              <Blueprint3.Icon
                                key="rightIcon"
                              />
                            </button>
                          </Blueprint3.Button>
                        </div>
                      </div>
                    </div>
                  </Transition>
                </CSSTransition>
              </div>
            </TransitionGroup>
          </Blueprint3.Overlay>
        </Blueprint3.Dialog>
      </Blueprint3.Alert>
    </AllowConnectionForDeviceAlert>
  </Router>
</BrowserRouter>
`;
